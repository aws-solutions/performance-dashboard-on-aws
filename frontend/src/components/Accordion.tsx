/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React, { ReactNode } from "react";

interface AccordionProps {
    children: ReactNode;
}

function Accordion(props: AccordionProps) {
    return <div className="usa-accordion">{props.children}</div>;
}

interface ItemProps {
    id: string;
    title: string;
    children: ReactNode;
    hidden?: boolean;
}

function Item(props: ItemProps) {
    return (
        <>
            <h2 className="usa-accordion__heading">
                <button
                    className="usa-accordion__button"
                    aria-expanded={props.hidden ? "false" : "true"}
                    aria-controls={props.id}
                >
                    {props.title}
                </button>
            </h2>
            <div
                id={props.id}
                className="usa-accordion__content usa-prose"
                hidden={props.hidden ? true : false}
            >
                <ul>{props.children}</ul>
            </div>
        </>
    );
}

Accordion.Item = Item;
export default Accordion;
