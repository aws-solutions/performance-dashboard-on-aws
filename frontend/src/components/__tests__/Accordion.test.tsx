/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import Accordion from "../Accordion";

test("renders an accordion with multiple elements", async () => {
    const wrapper = render(
        <Accordion>
            <Accordion.Item id="1" key="1" title="Element 1">
                <p>Hello this is an item in the accordion.</p>
            </Accordion.Item>
            <Accordion.Item id="2" key="2" title="Element 2">
                <p>Hello this is a second item in the accordion.</p>
            </Accordion.Item>
            <Accordion.Item id="3" key="3" title="Element 3">
                <p>Hello this is a third item in the accordion.</p>
            </Accordion.Item>
        </Accordion>,
    );
    expect(wrapper.container).toMatchSnapshot();
});
