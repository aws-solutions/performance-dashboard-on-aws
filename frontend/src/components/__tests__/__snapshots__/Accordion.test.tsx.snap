// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an accordion with multiple elements 1`] = `
<div>
  <div
    class="usa-accordion"
  >
    <h2
      class="usa-accordion__heading"
    >
      <button
        aria-controls="1"
        aria-expanded="true"
        class="usa-accordion__button"
      >
        Element 1
      </button>
    </h2>
    <div
      class="usa-accordion__content usa-prose"
      id="1"
    >
      <ul>
        <p>
          Hello this is an item in the accordion.
        </p>
      </ul>
    </div>
    <h2
      class="usa-accordion__heading"
    >
      <button
        aria-controls="2"
        aria-expanded="true"
        class="usa-accordion__button"
      >
        Element 2
      </button>
    </h2>
    <div
      class="usa-accordion__content usa-prose"
      id="2"
    >
      <ul>
        <p>
          Hello this is a second item in the accordion.
        </p>
      </ul>
    </div>
    <h2
      class="usa-accordion__heading"
    >
      <button
        aria-controls="3"
        aria-expanded="true"
        class="usa-accordion__button"
      >
        Element 3
      </button>
    </h2>
    <div
      class="usa-accordion__content usa-prose"
      id="3"
    >
      <ul>
        <p>
          Hello this is a third item in the accordion.
        </p>
      </ul>
    </div>
  </div>
</div>
`;
